pkg/skaffold/initializer/deploy/kustomize.go

```diff
@@ -38,28 +44,90 @@ func newKustomizeInitializer(kustomizations []string, potentialConfigs []string)
 		}
 	}
 	return &kustomize{
-		images:         images,
-		kustomizations: kustomizations,
+		defaultKustomization: defaultKustomization,
+		images:               images,
+		bases:                bases,
+		kustomizations:       kustomizations,
 	}
 }
 
 // deployConfig implements the Initializer interface and generates
 // a kustomize deployment config.
-func (k *kustomize) DeployConfig() latest.DeployConfig {
+func (k *kustomize) DeployConfig() (latest.DeployConfig, []latest.Profile) {
 	var kustomizeConfig *latest.KustomizeDeploy
-	// if we only have the default path, leave the config empty - it's cleaner
-	if len(k.kustomizations) == 1 && k.kustomizations[0] == deploy.DefaultKustomizePath {
-		kustomizeConfig = &latest.KustomizeDeploy{}
-	} else {
-		kustomizeConfig = &latest.KustomizeDeploy{
-			KustomizePaths: k.kustomizations,
+	var profiles []latest.Profile
+
+	// if there's only one kustomize path, either leave it blank (if it's the default path),
+	// or generate a config with that single path and return it
+	if len(k.kustomizations) == 1 {
+		if k.kustomizations[0] == deploy.DefaultKustomizePath {
+			kustomizeConfig = &latest.KustomizeDeploy{}
+		} else {
+			kustomizeConfig = &latest.KustomizeDeploy{
+				KustomizePaths: k.kustomizations,
+			}
 		}
+		return latest.DeployConfig{
+			DeployType: latest.DeployType{
+				KustomizeDeploy: kustomizeConfig,
+			},
+		}, nil
 	}
+
+	// if there are multiple paths, generate a config that chooses a default
+	// kustomization based on our heuristic, and creates separate profiles
+	// for all other overlays in the project
+	defaultKustomization := k.defaultKustomization
+	if defaultKustomization == "" {
+		// either choose one that's called "dev", or else the first one that isn't called "prod"
+		dev, prod := -1, -1
+		for i, kustomization := range k.kustomizations {
+			switch filepath.Base(kustomization) {
+			case "dev":
+				dev = i
+			case "prod":
+				prod = i
+			default:
+			}
+		}
+
+		switch {
+		case dev != -1:
+			defaultKustomization = k.kustomizations[dev]
+		case prod == 0:
+			defaultKustomization = k.kustomizations[1]
+		default:
+			defaultKustomization = k.kustomizations[0]
+		}
+		logrus.Warnf("multiple kustomizations found but no default provided - defaulting to %s", defaultKustomization)
```

Comment:
This doesn't seem like a warning, but I guess we filter out INFO by default.

Reply:
yeah, I wanted to make sure it finds its way to the user, because it's possible we end up putting their production overlay in as the default for dev.